backend/server.py

```
@@ -6194,12 +6194,41 @@ def _period_cutoff(period: str) -> datetime:
+                    docs.append(doc)
+            # Sort by timestamp
+            docs.sort(key=lambda d: str(d.get("timestamp", "")))
+        else:
+            # MongoDB: use the cursor with $gte
+            cursor = col.find({"timestamp": {"$gte": cutoff}}).sort("timestamp", 1)
+            if hasattr(cursor, 'to_list'):
+                docs = await cursor.to_list(length=100000)
+            else:
+                async for doc in cursor:
+                    docs.append(doc)
+    except Exception as exc:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)